src/gui/mzroll/projectdockwidget.h

```diff
@@ -31,8 +31,25 @@ class ProjectDockWidget : public QDockWidget
     QColor getSampleColor(mzSample* sample);
     QIcon getSampleIcon(mzSample* sample);
 
+    /**
+     * @brief Map stores the item along with its 
+     * checked and unchecked state.
+     */ 
+    map<QTreeWidgetItem*, bool> saveSampleState; 
```

Comment:
This should be a private member. Also give this a better name like `originalCheckStates` or something. I cannot tell what state it stores when reading the code without looking at the docstring.